lib/web/purchases_flutter_web.dart

```diff
@@ -0,0 +1,872 @@
+
+    // Check if SDK is already loaded
+    if (js.context.hasProperty('PurchasesHybridMappings')) {
+      js.context['purchasesLoaded'] = Future.value();
+    } else {
+      injectScript();
+    }
+  }
+
+  Future<dynamic> handleMethodCall(MethodCall call) async {
+    try {
+      switch (call.method) {
+        case 'setupPurchases':
+          return await _setupPurchases(call.arguments);
+        case 'setLogLevel':
+          return _setLogLevel(call.arguments);
+        case 'getCustomerInfo':
+          return _getCustomerInfo();
+        case 'getOfferings':
+          return _getOfferings();
+        case 'getCurrentOfferingForPlacement':
+          return _getCurrentOfferingForPlacement(call.arguments);
+        case 'getAppUserID':
+          return _getAppUserID();
+        case 'logIn':
+          return _logIn(call.arguments);
+        case 'logOut':
+          return _logOut();
+        case 'preload':
+          return _preload();
+        case 'purchase':
+          return _purchase(call.arguments);
+        case 'purchaseProduct':
+          return _purchaseProduct(call.arguments);
+        case 'purchasePackage':
+          return _purchasePackage(call.arguments);
+        case 'restorePurchases':
+          return _restorePurchases();
+        case 'close':
+          return _close();
+        case 'isAnonymous':
+          return _isAnonymous();
+        case 'isSandbox':
+          return _isSandbox();
+        case 'isConfigured':
+          return _isConfigured();
+        case 'syncPurchases':
+          return _syncPurchases();
+        case 'checkTrialOrIntroductoryPriceEligibility':
+          return _checkTrialOrIntroductoryPriceEligibility(call.arguments);
+        case 'invalidateCustomerInfoCache':
+          return _invalidateCustomerInfoCache();
+        case 'setAttributes':
+          return _setAttributes(call.arguments);
+        case 'collectDeviceIdentifiers':
+          return _collectDeviceIdentifiers();
+        case 'canMakePayments':
+          return _canMakePayments(call.arguments);
+        case 'setAllowSharingStoreAccount':
+          return _setAllowSharingStoreAccount(call.arguments);
+        case 'setSimulatesAskToBuyInSandbox':
+          return _setSimulatesAskToBuyInSandbox(call.arguments);
+        case 'setProxyURL':
+          return _setProxyURL(call.arguments);
+        case 'enableAdServicesAttributionTokenCollection':
+          return _enableAdServicesAttributionTokenCollection();
+        case 'presentCodeRedemptionSheet':
+          return _presentCodeRedemptionSheet();
+        case 'purchaseSubscriptionOption':
+          return _purchaseSubscriptionOption(call.arguments);
+        case 'purchaseDiscountedProduct':
+          return _purchaseDiscountedProduct(call.arguments);
+        case 'purchaseDiscountedPackage':
+          return _purchaseDiscountedPackage(call.arguments);
+        case 'getPromotionalOffer':
+          return _getPromotionalOffer(call.arguments);
```

Comment:
I'm thinking about joining all the cases that should no-op or return an unsupported error together here, instead of creating separate methods. I think that would improve visibility IMO.